prov/sockets/src/sock_progress.c

```diff
@@ -1429,7 +1429,6 @@ static int sock_pe_process_rx_send(struct sock_pe *pe, struct sock_rx_ctx *rx_ct
 			rx_entry->data = pe_entry->data;
 			rx_entry->ignore = 0;
 			rx_entry->comp = pe_entry->comp;
-			pe_entry->context = rx_entry->context;
```

Comment:
Why was this changed?

The change to setting data_len looks like a valid fix, versus a code cleanup.  The change to context needs more explanation.